runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/WindowingStrategyTranslation.java

```diff
@@ -365,35 +358,33 @@ public static SdkFunctionSpec toProto(WindowFn<?, ?> windowFn, SdkComponents com
 
   public static WindowFn<?, ?> windowFnFromProto(SdkFunctionSpec windowFnSpec) {
     try {
-      switch (windowFnSpec.getSpec().getUrn()) {
-        case GLOBAL_WINDOWS_FN:
-          return new GlobalWindows();
-        case FIXED_WINDOWS_FN:
-          StandardWindowFns.FixedWindowsPayload fixedParams = null;
-          fixedParams =
-              StandardWindowFns.FixedWindowsPayload.parseFrom(
-                  windowFnSpec.getSpec().getPayload());
-          return FixedWindows.of(Duration.millis(Durations.toMillis(fixedParams.getSize())))
-              .withOffset(Duration.millis(Timestamps.toMillis(fixedParams.getOffset())));
-        case SLIDING_WINDOWS_FN:
-          StandardWindowFns.SlidingWindowsPayload slidingParams =
-              StandardWindowFns.SlidingWindowsPayload.parseFrom(
-                  windowFnSpec.getSpec().getPayload());
-          return SlidingWindows.of(Duration.millis(Durations.toMillis(slidingParams.getSize())))
-              .every(Duration.millis(Durations.toMillis(slidingParams.getPeriod())))
-              .withOffset(Duration.millis(Timestamps.toMillis(slidingParams.getOffset())));
-        case SESSION_WINDOWS_FN:
-          StandardWindowFns.SessionsPayload sessionParams =
-              StandardWindowFns.SessionsPayload.parseFrom(windowFnSpec.getSpec().getPayload());
-          return Sessions.withGapDuration(
-              Duration.millis(Durations.toMillis(sessionParams.getGapSize())));
-        case SERIALIZED_JAVA_WINDOWFN_URN:
-          return (WindowFn<?, ?>)
-              SerializableUtils.deserializeFromByteArray(
-                  windowFnSpec.getSpec().getPayload().toByteArray(), "WindowFn");
-        default:
-          throw new IllegalArgumentException(
-              "Unknown or unsupported WindowFn: " + windowFnSpec.getSpec().getUrn());
+      String s = windowFnSpec.getSpec().getUrn();
+      if (s.equals(getStandardUrn(StandardWindowFnUrns.Enum.GLOBAL_WINDOWS))) {
```

Comment:
We could remove the case statements, but inlining getStandardUrn(...) is quite verbose here.